src/components/oInput.js

```diff
@@ -0,0 +1,227 @@
+import o from 'ojs-core';
+import isValidInputEventName from '../utils/isValidInputEventName';
+import updateComponentDb from "../utils/updateComponentDb";
+
+export default function oInput(typeOrConfig) {
+  if (!(this instanceof oInput)) {
+    return new oInput(typeOrConfig);
+  }
+
+  this.element = o('input');
+  this.input = this.element.element;
+  this.dbSet = false;
+  this.dbObject = null;
+  this.dbKey = null;
+  this.index = null;
+
+  if(typeof typeOrConfig === 'string') {
+    this.type(typeOrConfig);
+  }
+
+  //TODO: initConfigService
+  if(typeof typeOrConfig === 'object') {
+    // this.attr(typeOrConfig);
+  }
+}
+
+oInput.prototype.attr = function(attrs) {
+  this.element.attr(attrs);
+  return this;
+}
+
+oInput.prototype.type = function(type) {
+  this.element.attr({
+    type
+  });
+
+  return this;
+}
+
+oInput.prototype.name = function(name) {
+  if(typeof name === 'undefined') return this;//nie wiem czy sprawdzać też pusty string + inne walidacje
+  this.element.attr({
+    name
+  });
+  return this;
+}
+
+oInput.prototype.required = function(required) {
+  if(typeof required !== 'boolean') return this;
+
+  this.element.attr({
+    required
+  });
+  return this;
+}
+
+oInput.prototype.disabled = function(required) {
+  if(typeof disabled !== 'boolean') return this;
+
+  this.element.attr({
+    disabled
+  });
+  return this;
+}
+
+oInput.prototype.readOnly = function(readonly) {
+  if(typeof readonly !== 'boolean') return this;
+
+  this.element.attr({
+    readonly
+  });
+  return this;
+}
+
+oInput.prototype.id = function (id) {
+  if(typeof id === 'undefined') return this;//nie wiem czy sprawdzać też pusty string + inne walidacje
+
+  this.element.id(id);
+  return this;
+}
+
+oInput.prototype.onChange = function(event) {
+  if(typeof event !== 'function') return this;
+
+  this.element.event({
+    name: 'change',
+    fn: e => event(e, this)
+  });
+  return this;
+}
+
+oInput.prototype.onKeyUp = function(event) {
+  if(typeof event !== 'function') return this;
+
+  this.element.event({
+    name: 'keyup',
+    fn: e => event(e, this)
+  });
+  return this;
+}
+
+oInput.prototype.onKeyDown = function(event) {
+  if(typeof event !== 'function') return this;
+
+  this.element.event({
+    name: 'keydown',
+    fn: e => event(e, this)
+  });
+  return this;
+}
+
+oInput.prototype.onFocus = function(event) {
+  if(typeof event !== 'function') return this;
+
+  this.element.event({
+    name: 'focusin',
+    fn: e => event(e, this)
+  });
+  return this;
+}
+
+oInput.prototype.onFocusOut = function(event) {
+  if(typeof event !== 'function') return this;
```

Comment:
Może powinniśmy też jakieś warningu dawać, żeby to można było łatwiej debugować, że coś nie tak z argumentem, który podaliśmy. Bo to spoko, że nie wywali sie cały input jak zrobimy return this. Co o tym myślisz? Do przegadania